design-toolkit/components/src/components/table/table-body.tsx

```diff
@@ -10,17 +10,52 @@
  * governing permissions and limitations under the License.
  */
 
+import type { Row, RowData } from '@tanstack/react-table';
+import { useContext } from 'react';
+import { TableContext } from '.';
 import { tableBodyStyles } from './styles';
+import { TableRow } from './table-row';
 import type { TableBodyProps } from './types';
 
 export function TableBody({ className, ref, ...props }: TableBodyProps) {
+  const { getTopRows, getCenterRows, getBottomRows } = useContext(TableContext);
+
+  if (!getCenterRows.length) {
+    return <tbody>{props.children}</tbody>;
```

Comment:
Shouldn't this element include the same props (red, className, ...props) as the one in the second return?